utils/preferences/tests/dtf/data_provider.rs

```diff
@@ -0,0 +1,33 @@
+use super::*;
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{langid, LanguageIdentifier};
+
+struct DefaultPrefs {
+    pub und: DTFDefaultPreferencesBag,
+    pub list: &'static [(LanguageIdentifier, DTFDefaultPreferencesBag)],
```

Comment:
Question: These will come from the data provider, right? I assume this is obvious but I want to be explicit since my previous assumptions about this PR haven't always been correct.

I'm okay putting UND data in code, but nothing more than that.

Reply:
Yeah, I expect UND to also come from data provider.